File: evidence/generated-systems/angular/analytics-on-angular/src/app/models/AccessPolicy.js

// Define collection and schema for AccessPolicy
export interface AccessPolicy {
    name:
	type : string
    subjectName:
	type : string
    Workspace:
	type : Schema.Types.ObjectId
    Datasets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'DataSet' }]
    Dashboards:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Dashboard' }]
    Reports:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Report' }]
    Models:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Model_' }]
    FeatureSets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'FeatureSet' }]
    AccessLevel:
 	type : String
    SubjectType:
 	type : String
#
    collection: 'accessPolicys'
}

File: evidence/generated-systems/golang/analytics-on-golang/analytics-on-golang/internal/model/AccessPolicy.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// AccessPolicy Declaration
//==============================================================
type AccessPolicy struct {
    gorm.Model
     Name                                    string
    SubjectName                                    string
    WorkspaceId         *uint
    Workspace           *AnalyticsWorkspace `gorm:"foreignKey:WorkspaceId"`
     Datasets           []DataSet `gorm:"foreignKey:DatasetsFromAccessPolicyId"`
     Dashboards           []Dashboard `gorm:"foreignKey:DashboardsFromAccessPolicyId"`
     Reports           []Report `gorm:"foreignKey:ReportsFromAccessPolicyId"`
     Models           []Model_ `gorm:"foreignKey:ModelsFromAccessPolicyId"`
     FeatureSets           []FeatureSet `gorm:"foreignKey:FeatureSetsFromAccessPolicyId"`
    AccessLevel                      AccessLevel
    SubjectType                      SubjectType

// parent associations as their child

}

